server/model/status_page.js

```diff
@@ -98,6 +145,40 @@ class StatusPage extends BeanModel {
         return $.root().html();
     }
 
+    /**
+     * Get all data required for RSS
+     * @param {StatusPage} statusPage Status page to get data for
+     * @returns {object} Status page data
+     */
+    static async getRSSPageData(statusPage) {
+        // get all heartbeats that correspond to this statusPage
+        const config = await statusPage.toPublicJSON();
+
+        // Public Group List
+        const showTags = !!statusPage.show_tags;
+
+        const list = await R.find("group", " public = 1 AND status_page_id = ? ORDER BY weight ", [
+            statusPage.id
+        ]);
+
+        let heartbeats = [];
+
+        for (let groupBean of list) {
+            let monitorGroup = await groupBean.toPublicJSON(showTags, config?.showCertificateExpiry);
+            for (const a of monitorGroup.monitorList) {
+                const id = a.id;
+                const heartbeat = await R.findOne("heartbeat", "monitor_id = ? ORDER BY time DESC", [ id ]);
```

Comment:
FYI: This is fine here, because we cache the response for 5m, but usually this would be a N+1 query.
See https://github.com/louislam/uptime-kuma/pull/5025 for a realted PR.

I have not measured the performance impact of this PR on said endpoint as I currently am quite swamped with other work. If you can refactor this, that would be nice.
If you can't, so be it ^^

Reply:
You're fully right. To avoid the N+1, I guess I'd have to surgically pick out the code for groupBean.toPublicJSON and add the relevant info in a JOIN. Not too confident in this code base to do that 😄 

And as you say, the caching mitigates this issue a little.